src/components/extracrewdetails.tsx

```diff
@@ -37,9 +64,94 @@ class ExtraCrewDetails extends Component<ExtraCrewDetailsProps, ExtraCrewDetails
 		}
 
 		let self = this;
+		fetch('/structured/keystones.json')
+			.then(response => response.json())
+			.then(allkeystones => {
+				let crew_keystone_crate = allkeystones.find((k: any) => k.crew_archetype_id === this.props.crew_archetype_id);
+
+				if (crew_keystone_crate) {
+					self.setState({
+						keystone: {
+							name: crew_keystone_crate.name, flavor: crew_keystone_crate.flavor,
+							keystones: crew_keystone_crate.keystones.map((kid: any) => allkeystones.find((k: any) => k.id === kid))
+						}
+					});
+				}
+			});
+
 		fetch('/structured/crew.json')
 			.then(response => response.json())
 			.then(crew => {
+				// Find all crew who have any polestars in common
+				for (let i = 0; i < crew.length; i++) {
```

Comment:
```suggestion
				for (let i = 0; i < crew.length; i++) {
				  if (!crew[i].in_portal) continue;
```

Want to exclude crew not in portal, who can't be retrieved.